packages/aws-cdk-lib/core/lib/helpers-internal/cfn-parse.ts

```diff
@@ -386,98 +395,93 @@ export class CfnParser {
     }
   }
 
-  private parseCreationPolicy(policy: any, logicalId: string): CfnCreationPolicy | undefined {
-    if (typeof policy !== 'object') { return undefined; }
+  private parseCreationPolicy(policy: any, logicalId: string): FromCloudFormationResult<CfnCreationPolicy | undefined> {
+    if (typeof policy !== 'object') { return new FromCloudFormationResult(undefined); }
     this.throwIfIsIntrinsic(policy, logicalId);
     const self = this;
 
-    // change simple JS values to their CDK equivalents
-    policy = this.parseValue(policy);
-
-    return undefinedIfAllValuesAreEmpty({
-      autoScalingCreationPolicy: parseAutoScalingCreationPolicy(policy.AutoScalingCreationPolicy),
-      resourceSignal: parseResourceSignal(policy.ResourceSignal),
-    });
+    const creaPol = new ObjectParser<CfnCreationPolicy>(this.parseValue(policy));
+    creaPol.parse('autoScalingCreationPolicy', 'AutoScalingCreationPolicy', parseAutoScalingCreationPolicy);
+    creaPol.parse('resourceSignal', 'ResourceSignal', parseResourceSignal);
+    return creaPol.toResult();
 
-    function parseAutoScalingCreationPolicy(p: any): CfnResourceAutoScalingCreationPolicy | undefined {
+    function parseAutoScalingCreationPolicy(p: any): FromCloudFormationResult<CfnResourceAutoScalingCreationPolicy | undefined> {
       self.throwIfIsIntrinsic(p, logicalId);
-      if (typeof p !== 'object') { return undefined; }
+      if (typeof p !== 'object') { return new FromCloudFormationResult(undefined); }
 
-      return undefinedIfAllValuesAreEmpty({
-        minSuccessfulInstancesPercent: FromCloudFormation.getNumber(p.MinSuccessfulInstancesPercent).value,
-      });
+      const autoPol = new ObjectParser<CfnResourceAutoScalingCreationPolicy>(p);
+      autoPol.parse('minSuccessfulInstancesPercent', 'MinSuccessfulInstancesPercent', FromCloudFormation.getNumber);
+      return autoPol.toResult();
     }
 
-    function parseResourceSignal(p: any): CfnResourceSignal | undefined {
-      if (typeof p !== 'object') { return undefined; }
+    function parseResourceSignal(p: any): FromCloudFormationResult<CfnResourceSignal | undefined> {
+      if (typeof p !== 'object') { return new FromCloudFormationResult(undefined); }
       self.throwIfIsIntrinsic(p, logicalId);
 
-      return undefinedIfAllValuesAreEmpty({
-        count: FromCloudFormation.getNumber(p.Count).value,
-        timeout: FromCloudFormation.getString(p.Timeout).value,
-      });
+      const sig = new ObjectParser<CfnResourceSignal>(p);
+      sig.parse('count', 'Count', FromCloudFormation.getNumber);
+      sig.parse('timeout', 'Tiemout', FromCloudFormation.getString);
```

Comment:
```suggestion
      sig.parse('timeout', 'Timeout', FromCloudFormation.getString);
```

Perhaps we could have special method, built on top of `parse`, that converts the cases automatically, to reduce the risk of typos like this one.

```
parseConvertingCase('timeout', FromCloudFormation.getString);
```